rsconnect/actions.py

```diff
@@ -464,25 +464,40 @@ def validate_entry_point(entry_point, directory):
     return entry_point
 
 
-def locate_quarto(quarto):
+def which_quarto(quarto):
```

Comment:
I think I didn't properly get my point across here.

This needs a default argument, otherwise the fallback code is _never_ going to execute, because the function will err if called with no arguments.

Here's a REPL example, mirroring the structure of this function as written. If no default value is provided, the fallback code is inaccessible.

```
>>> def printdef(v):
...     if v:
...         print("v", v)
...         return
...     print("fallback")
...
>>> printdef(42)
v 42
>>> printdef()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: printdef() missing 1 required positional argument: 'v'
>>>
```

The default arg should be a falsy value like `""` or `None`.

```
>>> def printdef(v=""):
...     if v:
...         print("v", v)
...         return
...     print("fallback")
...
>>> printdef()
fallback
```
>>>

Reply:
That shouldn't matter — lines 478–499 are written expressly to allow the function to check well-known locations when no Quarto is supplied, and the function is available as part of the public API exported by the package.

```
>>> from rsconnect import actions
>>> actions.which_quarto()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: which_quarto() missing 1 required positional argument: 'quarto'
```

Reply:
Added a default value for the `quarto` argument as a stronger indication that `which_quarto` will discover a Quarto when one is not supplied.